### ISSUE TITLE
Regression: Notes created with API v0.2 no longer get titles

### ISSUE THREAD

**Issue by phedlund** [nextcloud/notes#538]
**Describe the bug**
Notes created with API v0.2 no longer get autogenerated titles. API v0.2 for creating notes did not support the `title` attribute. The title was autogenerated on the server.

**To Reproduce**
1. Call `POST /apps/notes/api/v0.2/notes` with body
```
{
"content": "My awesome note",
"category": "Top priority",
"modified": "12345555555",
"favorite": "false"
}
```
2. Inspect returned body
3. There is no title in the body

**Expected behavior**
A title in the response body

**Server (please complete the following information):**
- Notes app version: 3.5.0
- Nextcloud version: 18.0.5
- OS: Linux
- Web server: Apache
- PHP version: 7.2
- Database: Postgresql

**Client (please complete the following information):**
 - App: CloudNotes for iOS
 - OS: iOS

**Comment by korelstar** [comment]
Sorry, I can't reproduce this. How did you test it?

If I run the following command:

```
curl -u test:test -X POST -H "Content-Type: application/json" -d '{"content": "My awesome note","category": "Top priority","modified": 12345555555,"favorite": false}' http://localhost/index.php/apps/notes/api/v0.2/notes
```
I'm getting the following response:
```
{"id":6274,"title":"My awesome note","modified":12345555555,"category":"Top priority","favorite":false,"error":false,"errorMessage":"","content":"My awesome note"}
```

This looks correct to me.

Please note btw., that `modified`is of type `integer` and `favorite` is of type `boolean`. Therefore, values must not be in quotes. Especially for `favorite`, it looks like `"false"` is interpreted as `true`. Hence, please use `false` instead!

**Comment by phedlund** [comment]
OK, let me do some more investigating. I just know that code that my iOS client has used for years no longer works.

**Comment by korelstar** [comment]
There were many changes done recently (see https://github.com/nextcloud/notes/commits/master for details). I did some maintenance work (refactoring, improving code, ...) that *could* be the cause for the issue. But I also added a full API test that still executes successfully. Therefore, it is possible that something that was not intended did work, but now doesn't work.

Maybe the user can say at which time the problem was introduced, i.e. which Notes version did work  and which one not. Then we could look into the details.
